src/Engine/ProtoCore/Parser/atg/Start.atg

```diff
@@ -154,6 +154,9 @@ COMPILER DesignScriptParser
 		if (IsTypedVariable())
             return false;
 
+        if (la.value == "return")
+            return false;
```

Comment:
Does this mean that a `return` statement is considered an assignment?

Reply:
Ideally it shouldn't -- it should be a `return` statement. But our parser and codegen couple so tightly that is not easy to separate it out. Considering that we would have a unified parser, I made minimal change to the parser.